### PR TITLE
Add type definitions file index.d.ts

### PR DESCRIPTION
**Pull request by mstar-kk**
As described in the title, add type definitions file index.d.ts

### CODE REVIEW

File: index.d.ts [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+  buffer: Buffer,
+  base: 'dec' | 'hex' | 'bin' | 'oct',
+  options: formatOption
+): string;
+
+export function createBuffer(size: number): Buffer;
+
+export function createBufferFrom(object: object): Buffer;
+
+export function toDecimalString(buffer: Buffer, options: formatOption): string;
```
Comment (Luchanso) [inline index.d.ts:34]: nit: and this options too

**Review by mstar-kk — COMMENTED** [review]
My fault

File: index.d.ts (hunk 1 above)
Comment (mstar-kk) [reply index.d.ts]: Fixed

File: index.d.ts [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+  padstr?: string;
+  size?: number;
+  trim?: boolean;
+  prefix?: string;
+}
+
+export default function (
+  buffer: Buffer,
+  base: 'dec' | 'hex' | 'bin' | 'oct',
+  options: formatOption
```
[the remark this answers is not in the document]
Comment (mstar-kk) [reply index.d.ts]: Fixed

**Comment by Luchanso** [comment]
> My fault

LGTM (and work for me), but I think repo owner not maintained this package anymore 😞

**Review by Luchanso — APPROVED** [review]

**Comment by Luchanso** [comment]
I think one of the good solution create types in definitely typed (maybe I'll latter do it):

https://definitelytyped.org/guides/contributing.html
https://github.com/DefinitelyTyped/DefinitelyTyped

**Comment by mstar-kk** [comment]
> > My fault
> 
> LGTM (and work for me), but I think repo owner not maintained this package anymore 😞

but we can, I think this is the one of charm of open source, right?  :smile:

**Pull request merged**